src/client/cli/formatter/csv_formatter.cpp

```diff
@@ -55,8 +83,10 @@ std::string generate_instance_info_report(const mp::InfoReply& reply)
         "Name,State,Ipv4,Ipv6,Release,Image hash,Image release,Load,Disk usage,Disk total,Memory usage,Memory "
         "total,Mounts,AllIPv4,CPU(s),Snapshots\n");
 
-    for (const auto& info : mp::format::sorted(reply.detailed_report().details()))
+    for (const auto& info : mp::format::sort_instances_and_snapshots(reply.detailed_report().details()))
     {
+        assert(info.has_instance_info() &&
```

Comment:
The assert is correct here if it expresses a precondition of this function, which I think makes sense. But asserts are not meant for user messages or error handling (note that they are not even included in release builds). They are meant rather for programmers, to enforce invariants in the code, helping us catch mistakes (present or future). 

In this case, the assert signals that code calling this without enforcing that precondition is wrong. As it stands, that appears to be the case of the `info` command, IIANM. Whenever the user calls `mp info --format csv instance instance.snapshot`, we need to detect it there and fail early, preventing this from being called.

The alternative would be to throw here (instead of asserting), letting the exception bubble up to the user. However, that would require an unnecessary trip to the daemon and make for poorer UX.